Feature/story 133 Public landing page

priyankpat opened the pull request:
[Story 133 ](https://airtable.com/tblRjd2McKgRW9MKj/viwlRRNI0PZIxrGkc/recQ5HNQNv6ZcwUzX?blocks=hide)
[Figma](https://www.figma.com/file/v5HdxDsvO2NzMuZU6CArzF/MutualAid.world-Designs-in-progress?node-id=1810%3A21)
Also contain changes from #308.

All the content is static at the moment.

![image](https://user-images.githubusercontent.com/5585797/80051727-e9ea1c80-84e6-11ea-9e1e-afe4da1105b9.png)
![image](https://user-images.githubusercontent.com/5585797/80051819-1d2cab80-84e7-11ea-80a3-77210d889dde.png)

azmainamin commented:
Keep the about us page.

azmainamin commented:
@priyankpat If you want, you can close this PR and make a new one with the About Us page.

priyankpat commented:
Let's just keep this PR. Reverted the change to Aboutus. Let me merge changes from #308 with Appbar.

jwu910 commented on src/app/layout/Appbar/Appbar.test.js:
```diff
@@ -1,38 +1,70 @@
+  //   // Accessibility
+  //   expect(appBar.getByText("appbar test children")).toBeTruthy();
+  // });
 });
```
Are we commenting out this entire test suite to fix the tests?

priyankpat replied:
No we are not, I forgot to uncomment it. Push the changes with it enabled.

priyankpat commented:
Test failing due to Github service degradation. @jwu910 You will have to manually trigger the CI job.

priyankpat commented:
I can convert them to webp format. As for SVG, I agree with the use of MUI icons. I'll create an issue and new PR with the fix. Let's keep this PR separate.

azmainamin commented:
What's the situation with this PR?

priyankpat commented:
> What's the situation with this PR?

I don't think there are anymore changes to be made for this PR. As for the asset optimization, I will create separate issue and PR to go along with it after this one is merge.

jwu910 commented on src/app/page/Home/Home.js:
```diff
@@ -83,52 +362,41 @@ const HomePage = ({ auth, history }) => {
+            title="Need help?"
+            message="Sign up to request a food box, small errand, or a pharmacy pickup. You'll be matched with a
+            volunteer who will take care of you ASAP."
+            actions={[
+              {
+                label: "I Need Help",
+                onClick: () => null,
+              },
+            ]}
```
I dont think the pattern you've created here is the right direction for this.
We should use React APIs where available because it seems like you're re-creating something React already does for us.

The abstraction with a greetingCardComponent seems like the right direction. But I didn't see any actions props with more than one object in them.

This might be better served just passing a prop for a linkUrl and linkLabel if each card only has one call to action.

priyankpat replied:
Addressed in f8a481f. Removed the actions prop and simplified it to actionLabel, actionPress.

jwu910 commented on src/app/page/Home/Home.test.js:
```diff
@@ -46,7 +47,7 @@ describe("Home page", () => {
     expect(page.getByRole("main")).toBeInTheDocument();
     expect(page.getByTestId("btn-login")).toBeInTheDocument();
-    expect(page.getByTestId("btn-signup")).toBeInTheDocument();
+    expect(page.getByTestId("btn-donate")).toBeInTheDocument();
```
This test may need revision after the rest of the code changes above.

priyankpat replied:
Addressed in 5f11f14. Added additional test cases.

priyankpat commented on src/app/page/Home/Home.js:
```diff
@@ -83,52 +362,41 @@ const HomePage = ({ auth, history }) => {
+          <GreetingCardComponent
+            title="Need help?"
+            message="Sign up to request a food box, small errand, or a pharmacy pickup. You'll be matched with a
+            volunteer who will take care of you ASAP."
```
Addressed in b0620a4.

priyankpat commented on src/app/page/Home/Home.js:
```diff
@@ -56,19 +67,287 @@ const useStyles = makeStyles((theme) => ({
+        <H3 className={classes.ContactAdLabel}>To request food by phone,</H3>
+        <H3 className={classes.ContactAdLabel}>
+          call{" "}
+          <Link to="/" className={classes.ContactAdLink}>
```
The button is static at the moment since there is no information on how to handle the click action. Do we have enough data to populate the page?

priyankpat commented on src/app/page/Home/Home.js:
```diff
@@ -56,19 +67,287 @@ const useStyles = makeStyles((theme) => ({
+      <Body1 className={classes.DonateCardMessage}>
+        Many families can't afford fresh food. Please help sponsor food boxes for those in need.
+      </Body1>
+      <Button className={classes.DonateCardAction} onClick={() => null} data-testid="btn-donate">
```
See the comment below.

priyankpat commented on src/app/page/Home/VolunteerHome.jsx:
```diff
@@ -38,7 +38,9 @@ export default function VolunteerHome({ currentUser }) {
-    fetchAllAssociatedMissions();
+    if (!!currentUser && !!currentUser.uid) {
+      fetchAllAssociatedMissions();
+    }
```
It was causing Home.test to throw warnings due to the update not being in `act()`.

priyankpat commented on src/app/page/Home/Home.test.js:
```diff
@@ -59,7 +60,9 @@ describe("Home page", () => {
-    const page = renderComponent({ state });
+
+    let page = renderComponent({ state });
+
```
No reason, I forgot to undo the change when I was trying to fix a failing test. I reverted the change.

priyankpat commented on src/app/layout/Appbar/Appbar.js:
```diff
@@ -1,12 +1,23 @@
+  const isEmpty = useSelector((state) => state.firebase.auth.isEmpty);
+  function linkToAbout() {
+    window.location = "/about";
+  }
```
It is relevant to this PR.

priyankpat commented on src/app/layout/Appbar/Appbar.js:
```diff
@@ -18,6 +29,7 @@ const Appbar = ({ children }) => {
               <Logo title="MutualAidLogo" role="img" />
             </Link>
           </LogoContainer>
+          {isEmpty ? <AboutText onClick={linkToAbout}>About Resilience App</AboutText> : null}
```
I pulled the changes from #294. I fixed the anti-pattern by using Link in 
489c406. As for the auth check, not sure why it's being used. I removed it.

priyankpat commented:
> @priyankpat The scope of this ticket includes these images. I don't believe the images should be handled in another PR. The images of this size shouldn't be merged in.
> 
> Please let us know if you need help getting them web optimized.

I also changed the image assets to webp, removed svg and replaced them with material icon.

jwu910 commented on src/app/page/Aboutus/Aboutus.js:
```diff
@@ -1,46 +1,78 @@
 import React from "react";
+import { Link } from "react-router-dom";
 
+import AboutImage1 from "../../../img/AboutImage1.webp";
```
Asking for cross browser development requirements because according to [caniuse](https://caniuse.com/#feat=webp), webp image formats aren't available on safari and IE.

You can use web optimization tools to scale images down to lower resolution needed for web. PNG and JPG should be fine, so long as our image is scaled down for the views we need.

jwu910 commented:
@priyankpat Can you just rebase on latest and fix conflicts before we move forward? Thanks!

azmainamin commented:
@jwu910 Seems there are no merge conflicts? Should I merge this in?

jwu910 commented:
@azmainamin I think @priyankpat needs to resolve the conflicts and then we should be good
